dt-reports/magic-url-setup.php

```diff
@@ -82,6 +82,11 @@ private function add_app_row( $post_type, $app ){
         } else {
             $key = dt_create_unique_key();
             update_post_meta( get_the_ID(), $meta_key, $key );
+
+            // Ensure user meta options are also kept in sync.
+            if ( !empty( $record['corresponds_to_user'] ) ) {
```

Comment:
@kodinkat, this force enables all the apps.

Reply:
> @kodinkat, this force enables all the apps.

@corsacca - ensures whenever a new key is created and updated at the post meta level; it's also reflected at the user meta level too; which is queried within user management view.

Reply:
@kodinkat auto creating the key enables the app for the user.
We don't want this to happen automatically, this is why the user has the toggles.
Can you think of a good way around this issue?
In the end i'd rather remove them from this list then have them be auto enabled.

Reply:
@corsacca let me know how the latest tweaks holds up for you; which now ensures (in addition to correct permissions) there is a valid app key set in order to be displayed within record's frontend app list.

App keys are now solely handled within the user management section.